Source/ChilliSource/Rendering/Base/Renderer.cpp

```diff
@@ -43,55 +66,68 @@ namespace ChilliSource
     
     //------------------------------------------------------------------------------
     Renderer::Renderer() noexcept
-        : m_currentSnapshot(nullptr, Integer2::k_zero, Colour::k_black, RenderCamera())
+        : m_currentMainSnapshot(nullptr, Integer2::k_zero, Colour::k_black, RenderCamera())
+
     {
+
     }
     
     //------------------------------------------------------------------------------
     bool Renderer::IsA(InterfaceIDType interfaceId) const noexcept
     {
         return (Renderer::InterfaceID == interfaceId);
     }
+    
     //------------------------------------------------------------------------------
-    RenderSnapshot Renderer::CreateRenderSnapshot(const Integer2& resolution, const Colour& clearColour, const RenderCamera& renderCamera) noexcept
+    RenderSnapshot Renderer::CreateRenderSnapshot(const RenderTargetGroup* renderTarget, const Integer2& resolution, const Colour& clearColour, const RenderCamera& renderCamera) noexcept
     {
-        auto allocator = m_frameAllocatorQueue.Pop();
-        
-        return RenderSnapshot(allocator, resolution, clearColour, renderCamera);
+        return RenderSnapshot(renderTarget, resolution, clearColour, renderCamera);
     }
     
     //------------------------------------------------------------------------------
-    void Renderer::ProcessRenderSnapshot(RenderSnapshot renderSnapshot) noexcept
+    void Renderer::ProcessRenderSnapshots(IAllocator*&& frameAllocator, RenderSnapshot&& mainRenderSnapshot, std::vector<RenderSnapshot>&& offscreenRenderSnapshots) noexcept
```

Comment:
I'm pretty sure what this is doing is moving the pointer, so the caller has a pointer to an object and after calling this the pointer will be null. I'm doing this to explicitly denote that ownership of the allocator is being transferred into the renderer (which pushes the allocator back onto the queue when it is done with it). I could just have passed by value but that would have left a pointer on the caller side that they might be tempted to use